iree/tools/CMakeLists.txt

```diff
@@ -12,17 +12,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# bazel_to_cmake: DO NOT EDIT (Special logic is used throughout this file)
+
 add_subdirectory(test)
 
-# TODO: iree_select_target_platform macros.
-# TODO: skip targets disabled by options.
-iree_select_compiler_opts(IREE_HAL_DRIVER_MODULES
-  ALL
-    "iree::hal::dylib::dylib_driver_module"
-    "iree::hal::llvmjit::llvmjit_driver_module"
-    "iree::hal::vmla::vmla_driver_module"
-    "iree::hal::vulkan::vulkan_driver_module"
-)
+# Enable HAL driver modules based on options.
+set (_IREE_HAL_DRIVER_MODULES "")
+if(${IREE_HAL_DRIVER_DYLIB})
+  list(APPEND _IREE_HAL_DRIVER_MODULES iree::hal::dylib::dylib_driver_module)
+endif()
+if(${IREE_HAL_DRIVER_LLVM})
+  list(APPEND _IREE_HAL_DRIVER_MODULES iree::hal::llvmjit::llvmjit_driver_module)
+endif()
+if(${IREE_HAL_DRIVER_VMLA})
+  list(APPEND _IREE_HAL_DRIVER_MODULES iree::hal::vmla::vmla_driver_module)
+endif()
+if(${IREE_HAL_DRIVER_VULKAN})
+  list(APPEND _IREE_HAL_DRIVER_MODULES iree::hal::vulkan::vulkan_driver_module)
+endif()
```

Comment:
This is used several times. We could move this to one central position, maybe the main `CMakeLists.txt`?.

Reply:
Sure, de-duped.